lib/ansible/module_utils/common/arg_spec.py

```diff
@@ -268,9 +275,14 @@ def validate(self, parameters):
         result = super(ModuleArgumentSpecValidator, self).validate(parameters)
 
         for d in result._deprecations:
-            deprecate("Alias '{name}' is deprecated. See the module docs for more information".format(name=d['name']),
-                      version=d.get('version'), date=d.get('date'),
-                      collection_name=d.get('collection_name'))
+            if 'name' in d:
```

Comment:
One thing I'm not really happy about is that `name` is used as an indicator that it is an alias deprecation. I would probably replace `name` by `alias`, but that would be a larger change that shouldn't happen in this PR IMO.

Reply:
It doesn't look like `name` is used for anything other than aliases - `msg` is used for deprecated parameters. Renaming it (or consolidating with msg) would be nice.

Reply:
I've started working on a PR to improve this, and then I noticed that there are probably a lot more bugs in there when it comes to suboptions (and using aliases for the option that you have suboptions of), both for parameter deprecation and for alias deprecation. This will be fun...

Reply:
Actually parameter deprecation works fine, it's alias deprecation that only work on toplevel.

Reply:
I tried to fix some long-term issues with suboptions in https://github.com/ansible/ansible/pull/76578/files too. I never got a review on that, so if you spot any more related issues I'd be happy to fix/review yours.